.jenkinsfile

```diff
@@ -1,113 +1,243 @@
+                    // volume on the filesystem which is shared by all docker instances. Since the docker executor
+                    // doesn't know the other running instances it cannot locking the workspace.
+                    ws("$env.SEQAN_BASE_DIR/ws/$env.JOB_NAME/$env.DOCKER_SLAVE_NAME/$label")
+                    {
+                        stage('install')
+                        {
+                            sh """
+                                sudo apt-get update
+                                sudo apt-get install -y $axis_compiler_value cmake git zlib1g-dev libbz2-dev
+                            """
+
+                        }
+                        run_unit_tests(env.DOCKER_SLAVE_NAME,
+                                       axis_compiler_value,
+                                       axis_build_type_value,
+                                       test_model,
+                                       axis_cereal_value)
                     }
                 }
             }
         }
-        stage ('Deploy') { // conditional deploy step.
-            agent none
-            when { // only execute if nightly build on 'master' branch or if seqan was tagged with a new release.
-                beforeAgent true
-                anyOf {
-                    allOf {
-                        branch 'master'
-                        environment name: 'BUILD_TRIGGER', value: 'cron'
-                    }
-                    // tag "seqan-*"  // TODO: Currently breaks the pipeline: see https://issues.jenkins-ci.org/browse/JENKINS-52554
-                }
+    }
+    tasks["ubuntu-${axis_compiler_value}"] =
+    {
+        parallel subTasks
+    }
+}
+
+
+/* sanitizer builds
+ * g++-7 sanitizer options
+ */
+
+/* doc builds
+ * user develop
+ */
+
+/* coverage build
+ * g++-8 debug mode
+ */
+
+/* valgrind build
+ *
+ */
+
+/* clang-tidy
+ *
+ */
+
+/* performance build
+ * g++-7, g++-8 [, g++-9] [, clang-concpets]
+ */
+
+// ============================================================================
+// Execute build matrix
+// ============================================================================
+
+stage ("matrix")
+{
+    parallel tasks
+}
+
+// ============================================================================
+// Cleanup steps executed in the nightly builds
+// ============================================================================
+
+if (is_nightly)
+{
+    stage ("cleanup")
+    {
+        // On the docker nodes we need to regularly check if the a pull request was closed in order
+        // to remove the workspace which is stored persistently on the docker server.
+        node("ubuntu_1604 || ubuntu_1804")
+        {
+            // Get all open PRs and scan their PR number
+            String pulls = sh(script: "curl -X GET https://api.github.com/repos/seqan/seqan3/pulls", returnStdout: true).trim()
+            def change_ids = pulls.findAll(/number": (\d+),/)
```

Comment:
This looks kind of hacky :see_no_evil:

Reply:
Oh, I which I could have a done it differently (see my issue: https://issues.jenkins-ci.org/browse/JENKINS-56111) But it does not work or I just couldn't figure out how to enable this.